### PR TITLE
zephyr: Enable support for RISCV

### PR DESCRIPTION
**Pull request by carlocaione**
Make libmetal usable also for RISCV targets.

Signed-off-by: Carlo Caione <ccaione@baylibre.com>

### CODE REVIEW

File: lib/system/zephyr/riscv/CMakeLists.txt [hunk 1]
```diff
@@ -0,0 +1,5 @@
+collect (PROJECT_LIB_HEADERS sys.h)
+
+collect (PROJECT_LIB_SOURCES sys.c)
+
+# vim: expandtab:ts=2:sw=2:smartindent
```
Comment (arnopo) [inline lib/system/zephyr/riscv/CMakeLists.txt:5]: useless line to clean up

**Review by arnopo — COMMENTED** [review]
minor clean-up else LGTM

File: lib/system/zephyr/riscv/CMakeLists.txt (hunk 1 above)
Comment (carlocaione) [reply lib/system/zephyr/riscv/CMakeLists.txt]: done

**Review by arnopo — APPROVED** [review]

**Review by edmooring — APPROVED** [review]
Looks good to go.

**Comment by edmooring** [comment]
Can we add at least a build test to the CI?

**Comment by arnopo** [comment]
> Can we add at least a build test to the CI?

That's a good point! at least we could add build test with RISC compiler in CI.
@carlocaione 
I suppose that everything is available to build for RISCV in Zephyr environment
what should be update in [Zephyr build sequence](https://github.com/OpenAMP/libmetal/blob/main/.github/actions/build_ci/entrypoint.sh#L55) to build for RISCV?

**Comment by carlocaione** [comment]
> what should be update in [Zephyr build sequence](https://github.com/OpenAMP/libmetal/blob/main/.github/actions/build_ci/entrypoint.sh#L55) to build for RISCV?

Only one line.

For RISCV64:
```
cmake . -DWITH_ZEPHYR=on -DBOARD=qemu_riscv64 -DWITH_TESTS=on -Bbuild-zephyr
```
for RISCV32:
```
cmake . -DWITH_ZEPHYR=on -DBOARD=qemu_riscv32 -DWITH_TESTS=on -Bbuild-zephyr
```

**Comment by arnopo** [comment]
> > what should be update in [Zephyr build sequence](https://github.com/OpenAMP/libmetal/blob/main/.github/actions/build_ci/entrypoint.sh#L55) to build for RISCV?
> 
> Only one line.
> 
> For RISCV64:
> 
> ```
> cmake . -DWITH_ZEPHYR=on -DBOARD=qemu_riscv64 -DWITH_TESTS=on -Bbuild-zephyr
> ```
> 
> for RISCV32:
> 
> ```
> cmake . -DWITH_ZEPHYR=on -DBOARD=qemu_riscv32 -DWITH_TESTS=on -Bbuild-zephyr
> ```

Thanks!
I sent a related pull request: https://github.com/OpenAMP/libmetal/pull/195

**Pull request merged**